src/tests/read_test.cpp

```diff
@@ -0,0 +1,75 @@
+
+#define BOOST_TEST_MODULE ASTE
+#define BOOST_TEST_DYN_LINK
+#include <boost/test/unit_test.hpp>
+
+#include <iostream>
+#include <mesh.hpp>
+#include <numeric>
+#include <string>
+#include <vector>
+
+struct Case {
+  std::string fname{};
+  std::string dataname{};
+  int         dim{};
+};
+
+std::vector<Case> Cases;
+
+BOOST_AUTO_TEST_SUITE(read_test)
+
+BOOST_AUTO_TEST_CASE(read_test)
+
+{
+  //Add Cases Here
+  Cases.push_back(Case{"../src/tests/reference_scalars", "Scalars", 1});
+  Cases.push_back(Case{"../src/tests/reference_vector2d", "Vector2D", 2});
+  Cases.push_back(Case{"../src/tests/reference_vector3d", "Vector3D", 3});
+
+  for (auto current_case : Cases) {
```

Comment:
We should split these into different tests.
You can write a `void read_test(const Case& case)` and have individual tests call this function:
```cpp
void read_test(const Case& case)
{
 BOOST_TEST(something);
}

BOOST_AUTO_TEST_CASE(Scalars)
{
  read_test(Case{"../src/tests/reference_scalars", "Scalars", 1});
}

BOOST_AUTO_TEST_CASE(X)
{
  read_test(Case{"../src/tests/X", "X", 1});
}
```